main/zoe/guide/contracts/multipoolAutoswap.md

```diff
@@ -0,0 +1,169 @@
+secondary token. Secondary tokens can be exchanged with each other, but only through
+the central token. For example, if X and Y are two token types and C is the central
+token, a swap giving X and wanting Y would first use the pool (X, C) then the pool
+(Y, C). There are no direct liquidity pools between two secondary tokens.
+
+There should only need to be one instance of this contract, so liquidity can be
+shared as much as possible. Each secondary currency has a separate pool of liquidity.
+
+When the contract is instantiated, the central token is specified in the
+terms. Separate invitations are available by calling methods on the publicFacet for
+adding and removing liquidity and for making trades. Other publicFacet operations
+support querying prices and the sizes of pools. New Pools can be created with
+`addPool()`.
+
+When making trades or requesting prices, the caller must specify that either the
+input price (swapIn, getInputPrice) or the output price (swapOut, getOutPutPrice) is
+fixed. For swaps, the required keywords are `In` for the trader's `give` amount, and
+`Out` for the trader's `want` amount.  `getInputPrice()` and `getOutputPrice()` each
+take an amount for the currency that is being specified, and a brand to indicate the
+currency of the desired amount.
+
+When adding and removing liquidity, the keywords are `Central`, `Secondary`, and
+`Liquidity`. Adding liquidity uses `Central` and `Secondary` in the `give` section,
+while to remove liquidity the `give` section only needs the particular liquidity
+token specified for `give`, and it will figure out the appropriate amounts to pay
+out.
+
+Transactions that don't require an invitation include `addPool()` and the queries
+(`getInputPrice()`, `getOutputPrice()`, `getPoolAllocation()`,
+`getLiquidityIssuer()`, and `getLiquiditySupply()`).
+
+This contract follows the design of [UniSwap](https://uniswap.org/).
+
+## The MultipoolAutoswap API
+
+These examples presume that the central token is bucks. Moola and simoleans are
+secondary currencies.
+
+### Trading with MultipoolAutoswap
+
+Once trading pools have been set up, a new trader can interact by asking for the
+current price, making an invitation, and making an offer. If Sara needs 275 moola for
+a deal she has negotiated and has simoleans she can spend, she can use
+getOutputPrice() to get a quote.
+
+```js
+const quote = E(publicFacet).getOutputPrice(
+  moolaAmountMath.make(275), simoleansBrand);
+  ```
+  
+Let's assume the quote is for 216 simoleans. Sara believes the price is somewhat
+volatile, and she doesn't want to make repeated calls, so she pads her offer. If the
+appropriate pools don't exist, she'll get an error (`brands were not recognized`).
+
+```js
+const saraProposal = harden({
+  want: { out: simoleans(275) },
+  give: { in: moola(220) },
+});
+
+const swapInvitation = await E(publicFacet).makeSwapOutInvitation();
+const moolaPayment = harden({ In: sMoolaPurse.withdraw(220) });
```

Comment:
```suggestion
const moolaPayment = harden({ In: sarahMoolaPurse.withdraw(moola220) });
```
`s` alone looked like a typo. Also can't withdraw a value

Reply:
Should be "Sara" for consistency with the text.